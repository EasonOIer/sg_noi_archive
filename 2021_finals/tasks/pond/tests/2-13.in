1951 975
999802 999317 998636 998029 997995 997792 996325 995259 995064 994660 993116 991744 991437 990602 990333 988963 988630 988512 988355 986721 985817 985579 985491 984905 984431 983792 983752 980685 980274 978730 978594 977647 976802 976790 975900 975017 973222 971162 970196 969728 968648 967173 30125 965884 965682 965532 964923 961853 961781 958375 957930 957667 956231 954154 953100 953040 952005 951141 950985 949299 948275 948221 945408 943525 942761 941406 940983 940716 940465 938857 938722 937077 936332 936258 936030 934422 932587 928195 925736 924918 924296 922255 921039 920552 920146 918825 918303 917423 913388 912951 912372 910752 909965 909843 908480 907648 907289 907250 905196 903007 900907 900858 900003 897947 897573 896876 895408 893015 891837 891436 890877 888556 888373 883754 883364 882145 882044 880623 880136 879780 878670 878315 877645 876973 873747 873167 873012 872971 872229 870607 869228 868677 868551 866966 866621 866514 866240 863129 862907 862078 861871 861680 861623 859819 859664 859473 858590 857140 856253 856182 856153 856042 855470 854868 854119 854025 853973 852622 851823 851335 850507 849816 848045 845753 845619 845034 841644 841572 840727 838578 838115 836637 834061 833731 833683 832932 829074 827634 826578 822630 822584 822559 822412 821225 820112 819772 818623 814428 812217 811938 811827 809461 809378 807282 806372 805226 805191 804094 802730 802026 799831 799795 798730 797605 796820 796453 795905 791943 789016 786619 784042 781261 776925 776687 775932 775620 772363 770877 770267 770255 769417 767544 767277 766649 765653 763442 763307 763082 762891 761781 760911 759320 758509 756785 756724 755816 755358 753350 752906 752058 750910 750260 749086 747549 746449 746370 741536 739580 738684 737306 736230 736117 734516 733511 732802 731923 731647 731549 731211 730718 725652 724998 722836 719534 719467 714706 714458 712169 708973 708758 708316 707954 707682 707375 706821 706274 705892 704488 704322 704085 702937 702607 702483 702399 699965 699225 697482 695928 695097 694665 694335 689225 687982 687457 687048 684935 684785 684404 682942 681174 681025 679372 679222 678948 678605 678602 677324 672369 669717 669338 669193 668452 667834 665772 663928 662394 662191 662013 661554 661427 661124 660624 659735 658926 654511 651131 650304 650304 650165 649754 648952 648372 648286 648058 648014 646863 646103 645337 644696 643155 639589 639129 637623 637424 636884 636500 636255 635623 634845 630034 628810 628671 628234 627461 627234 626370 625809 624252 623358 622943 622877 621006 620765 617741 616896 615585 615314 611938 610477 608242 608042 607561 607149 607143 607118 603764 602640 601937 601899 601850 601753 601112 601031 599621 599604 599419 599344 598359 597879 597768 597433 595560 593162 592924 591109 590946 587894 587686 587529 586922 585983 585816 584270 583534 580479 579432 576852 575998 574626 573978 573796 572361 572206 571783 571392 570037 569871 569532 569145 569086 565185 565118 564930 564142 563964 561894 560751 560502 560101 559220 558626 554180 552799 552356 551093 550152 548814 547007 546691 546172 545453 542022 538171 536866 535453 535332 533350 530676 530484 528458 528066 527642 526352 526093 525327 524994 524458 523540 522353 520844 518705 517646 516402 515106 514275 514014 513309 513018 511135 510948 507512 506526 506028 505434 505237 504597 503839 502366 501399 500764 500481 500211 498592 498143 497724 497213 497111 494728 493828 490127 488948 488649 486710 483656 481965 481130 481111 480446 480280 479890 479515 478876 478811 477403 967146 475985 472472 471602 471398 470374 469855 468901 468528 466768 465652 465576 462711 462110 461596 460604 458340 457500 457446 455691 455316 455132 454847 452166 450655 449419 447628 446033 441986 440614 439036 438926 435565 435558 434518 433409 432840 431684 431080 429407 428298 427583 265775 426495 426067 425167 424349 423602 423336 423244 422423 420334 419276 418238 417918 414462 413730 413591 413381 412201 408770 407570 407296 404668 404649 403881 403279 402667 402595 401574 400898 398706 397455 397418 395833 392746 390939 388355 387224 385841 385500 383958 383218 381427 377556 376118 374889 374515 373621 373569 373061 371555 369086 368921 367623 364306 361759 361756 360894 358341 357443 726045 356149 355417 352676 352209 350267 350222 349460 348481 348012 345553 344880 344307 343997 342245 341984 341657 340446 340314 339791 339323 339090 336110 335017 333446 332507 332191 332187 331896 331782 330840 328950 328527 328396 327306 327218 326091 325110 323849 323401 322842 320951 318779 318762 317118 316924 316922 316883 316534 316472 316428 314726 313297 313238 312425 311184 311146 311078 311034 310976 309954 309016 308341 308286 306732 306387 306179 305134 304757 304705 304107 303327 302426 302383 302315 302029 302002 301665 301377 300368 300070 299295 298732 298277 297463 294535 294178 294127 293755 291683 290897 289009 287707 287617 286376 285887 285688 284088 282714 282070 281610 278969 276493 275966 271514 271228 268075 267557 267156 267036 266866 266852 426734 264072 263863 263782 263409 262958 262844 261579 259221 258928 258532 257895 257498 255211 254338 253456 253286 251965 251901 251455 249807 249334 248934 247681 247263 244008 242562 241729 241635 240392 239612 237614 235305 234315 232645 232176 229597 228026 227320 227282 224784 222534 220987 218587 217689 217620 217410 216116 216028 215124 214961 214196 212452 212044 211481 211105 209051 209011 208614 207768 96144 205964 205378 204113 204030 163218 202314 201862 201187 198964 197154 196792 948729 188476 188054 187860 184452 183218 182197 181521 178573 178254 176093 174643 173089 171640 171470 170636 170028 169674 168971 166736 165731 164880 203273 162800 159909 158222 158096 156492 156182 154387 152266 150974 149173 149073 148832 148578 146967 144972 143484 140600 139576 139083 138940 138058 136277 135180 134683 134086 132729 131504 130335 130189 129356 127401 127371 127345 127276 125280 123466 123407 121121 120679 119494 118322 117202 116342 114715 114516 113273 111141 110951 109385 109306 108859 108103 107473 107011 106991 106639 106624 106561 103535 102043 99927 99773 99274 98647 206185 94848 94611 94334 93893 92375 91024 89968 89509 87891 137146 87246 86928 86142 86051 85599 84852 84497 84177 82476 81392 80639 80561 79829 77846 76941 72566 71623 70915 69866 69858 67710 65116 63150 60763 60655 59920 59880 59077 58570 57805 57258 54869 53705 51915 50923 50801 50614 50331 48886 45506 44856 44457 44381 43826 43259 42504 42274 42159 38818 38715 38132 37968 37537 37340 37307 35117 32134 30582 476489 29464 27832 27443 25412 20219 19103 18167 18134 17270 17237 15028 14980 14213 11678 10192 9004 9001 8415 7222 7077 6029 5908 4362 3364 2698 1380 1113 1036 121 1189 1736 3711 5487 7032 7376 7803 8529 8605 11003 11128 12668 13230 13238 13793 13870 14763 15853 16337 20025 21593 21998 22825 23745 25330 25549 25588 25745 26597 26741 28429 28648 28978 30162 32713 33755 34138 35001 35953 36056 36197 37707 38845 40193 41340 41699 42967 43042 43207 44338 44927 46177 46946 47436 47710 48595 48925 52081 53652 53814 53933 55350 55713 56445 56952 57604 59566 61545 62507 63158 63692 63937 65787 67451 67878 70144 72517 73106 75356 75694 76200 76355 77580 78313 78631 79643 81925 84134 84490 86128 86630 87855 89120 89631 90359 90535 90873 91289 92158 93120 94119 94600 96155 96431 97620 99219 102648 103121 103258 105200 107009 107084 110550 110608 111605 111666 111972 115265 115589 116082 116282 117045 117367 118019 118337 120861 123456 124352 125929 127127 128478 129331 129596 129913 131180 131432 131473 131562 131651 132217 132290 134830 135064 135635 136545 87845 143429 143442 143841 144525 146516 148151 151471 153440 153498 154547 154833 155165 158410 160675 160729 161326 162043 164383 165181 167875 168054 168556 168725 169702 169990 170206 172402 175488 175820 175876 176545 177460 178110 178571 179182 180702 180872 181436 183329 184070 184607 184834 189549 192813 197973 198842 199772 200479 200980 201522 202272 202312 202396 204866 206028 206138 206650 206828 208750 208838 209656 209883 212638 407171 212697 213265 213655 213802 216523 216969 218691 219530 220423 221734 222173 223921 226975 227314 229552 232102 233407 236174 237447 237906 240928 240937 241620 241958 244090 244336 244946 245232 245521 247422 248005 249409 250563 251707 252439 254233 254248 256250 258378 258595 258632 258653 259510 259549 260459 260914 261310 262299 262781 263132 264326 264566 264620 264805 265438 266020 267817 268651 270935 271317 272644 275120 276730 279125 282022 282814 284275 285993 286333 286834 286894 288091 290805 291707 292598 292620 293402 294097 294367 294863 295158 295259 295843 299730 300191 300919 301568 302199 302990 303030 303215 303557 307903 309604 309885 310270 311573 312800 313057 313888 314988 316795 317267 318027 318870 320528 321016 321575 321737 321936 322658 323775 323965 325493 326149 326787 327319 327538 331556 332173 332245 333211 334017 334385 334689 335698 337587 338243 339905 340517 340878 340898 341468 342185 342598 342660 344125 345364 347790 348108 349052 350534 350572 351996 352877 352899 356242 358165 358176 358894 358904 359210 359606 359826 359901 361005 363473 364743 365442 365572 365850 368346 369410 371565 373627 375244 376162 378374 379460 379613 380325 380386 381024 381123 382108 383235 383582 384768 384884 385005 386333 386666 387104 388909 389376 392010 392605 394628 395103 395762 395818 397128 397917 399812 400849 402155 403432 403467 404172 405014 406363 406437 212648 408172 409022 417511 421489 422933 424909 425868 425869 428090 430103 431167 433189 434518 435135 435840 436330 437565 438789 440545 440730 440810 443454 444149 445012 445993 447728 448073 448813 448944 451579 451693 451986 454528 456107 457731 457794 459216 460975 461253 463345 463452 464819 466494 466721 466878 467027 468006 468309 469610 471863 475208 475540 475896 476373 477308 478074 478808 483217 483298 483667 483828 484116 485518 486275 486467 486664 487102 489994 490601 491239 492433 492885 493141 493686 494688 495291 496430 497623 498028 500452 503346 503769 504788 505566 505829 506330 506819 508573 509095 511091 513416 516260 516829 517045 517828 519876 520973 521766 522550 523852 525441 525477 525662 527703 530408 530539 532166 536819 538557 538994 540782 540845 541132 541289 545834 547137 547644 548599 549296 550879 550911 551912 552357 552397 552658 553140 553844 554022 554623 554743 555686 556146 556634 557304 558115 559898 561901 562079 563168 563329 563670 563936 565835 566597 566767 566886 568269 568423 570327 572141 572513 572579 574030 574878 577831 578029 580293 581097 581727 583942 583995 584189 587998 588883 589239 590091 590186 591258 591320 591681 592439 592515 593263 593975 594123 595840 596618 597006 597519 599269 600378 604746 605847 606442 606812 608450 609288 610426 614747 615828 617421 621627 623452 624097 624270 624613 625581 626005 627938 628046 628411 628672 631378 632145 633371 634005 635782 637466 637508 638619 638826 638969 639052 639429 639431 639638 642874 644316 645285 645567 647104 647343 647895 647951 648501 649640 649983 650534 650738 651468 651817 652112 652291 652353 655905 656816 657299 659182 660166 662342 663524 663869 664163 664187 665101 665153 665383 665517 665579 667149 667392 667707 668769 669483 669664 669979 671733 672608 672620 674018 674381 675817 676281 679744 679870 681344 681375 681419 681589 681785 688981 691349 691879 692088 692696 694015 695815 696534 697990 698826 699064 700814 701784 702879 705359 705386 705845 706141 706756 707720 709456 710291 711565 712163 712408 712982 716036 716394 717476 717852 719146 719384 720736 721219 722605 722693 724220 356336 727491 728170 729875 730566 730798 732106 733645 733886 734897 734996 735346 735716 736366 736852 737956 738702 739880 741832 743099 744778 747123 748231 749906 749962 750688 752949 753591 755035 755234 756333 759436 759633 761949 765106 765135 767139 768746 768823 193565 769241 770805 771854 772379 772855 774286 774599 775555 776123 776160 776171 778112 779491 781293 781573 782654 784467 784700 785788 785933 786513 787039 787454 788235 790115 790402 791162 792606 792635 792794 792965 793396 795157 795167 800754 802108 805475 805861 807945 807996 808029 808549 809596 810373 811679 811931 812267 813547 814801 816144 816959 817359 818536 819018 819849 821670 823002 823805 826712 828205 829572 829870 831104 832100 832962 833865 835747 835853 837509 838809 838847 839122 840092 840349 841487 842441 844859 844915 845380 845412 847414 848136 848728 849591 850098 851263 853475 855920 858183 859023 859460 862212 865484 865557 866075 867726 868134 869219 869295 869316 870993 871529 873409 873713 874445 874584 874623 875172 875392 877597 877797 879967 880036 881420 883228 884666 889000 889414 892408 892536 893018 893890 895308 895673 899957 902195 902369 904308 908115 911890 914709 916896 918649 918706 919635 921798 922832 923264 926808 930362 933749 934097 934479 934557 934691 935913 936174 938343 938417 940100 940528 941614 941980 942985 943831 943905 944791 945154 945612 945727 946851 947450 948682 768833 951010 953319 954245 954619 954996 954997 957026 957344 958276 958467 959753 960139 961337 964111 964731 965881 967866 967905 968457 968498 968963 969638 970192 970642 973630 974060 974745 975585 976248 979063 979479 979890 980011 981212 981794 982687 982790 984624 985077 989033 990314 992113 992527 993175 994736 995868 996262 996420 996991 998987 1000000
