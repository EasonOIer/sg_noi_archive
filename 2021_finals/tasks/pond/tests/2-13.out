1114325599299